newsrevenuehub/rev-engine
# NRH-175 - Org admins can delete donation pages from SPA

#### What's this PR do?

Adds a delete icon to the donation page editor. When clicked, if the page is unpublished, it will be deleted, and user will be redirected to the pages list view in dashboard.

If the page is published, the user will see a confirmation modal that explains that they are deleting a page that is published, and they they have to confirm or back out. If they confirm, the page deletes, and they are redirected to the pages list view. If they decline, they go back to the page editor view.

This PR also adds soft deletion to the DonationPage model. Hub admins will have the option to restore soft-deleted donation pages from the Django admin, if that is required.

#### How should this be manually tested?

Log in to spa as org user. Create one page and don't publish it. Create another and publish it. Go to the page editor view for each page, and confirm behavior described in previous section for published vs. unpublished page

#### Do any changes need to be made before deployment to staging? production? (adding environment variables, for example)?

No.
